cmd/server-main.go

```diff
@@ -972,11 +972,37 @@ func serverMain(ctx *cli.Context) {
 	}
 
 	// Initialize users credentials and policies in background right after config has initialized.
+	globalIAMFullyInitialized.Store(true)
 	go func() {
 		bootstrapTrace("globalIAMSys.Init", func() {
 			globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
 		})
 
+		// Start a thread to keep checking IAM initialized and re-trigger
+		go func() {
```

Comment:
What is the point of this goroutine? We are already in a goroutine that is terminated as soon as this has been spawned.

Reply:
I thought these routines are disjoint in nature and this inner would keep running the ticker and wait for Init to succeed and then set console config fresh. Actually while testing it worked like that where the openid endpoint was down when I started MinIO and this ticker kept waiting. Once I started openid endpoint, it succeeded the Init and then console srarted allowing loin with openid.
Kindly guide if this understanding is wrong.

Reply:
ok, let me try without async

Reply:
I tried making this non async as below
```
            {
			ticker := time.NewTicker(10 * time.Second) // Check every 10 seconds
			defer ticker.Stop()
			for {
				select {
				case <-ticker.C:
					if !globalIAMFullyInitialized.Load() {
						globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
						if globalIAMFullyInitialized.Load() && globalBrowserEnabled {
							bootstrapTrace("initConsoleServer", func() {
								srv, err := initConsoleServer()
								if err != nil {
									logger.FatalIf(err, "Unable to initialize console service")
								}

								setConsoleSrv(srv)
							})
							return
						}
					}
				}
			}
		}
```
but in this case it blocks and console doesnt get initialized. Need is that even if openid not reachable keep trying to connect to it in background but at the same time go ahead and allow login using normal creds.

On top of recent updates on PR below change still keeps it working

```
diff --git a/cmd/iam.go b/cmd/iam.go
index 9d1c15a19..6ed42dd76 100644
--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -371,7 +371,9 @@ func (sys *IAMSys) Init(ctx context.Context, objAPI ObjectLayer, etcdClient *etc
 
        refreshInterval := sys.iamRefreshInterval
 
-       go sys.periodicRoutines(ctx, refreshInterval)
+       if localIAMInitialized {
+               go sys.periodicRoutines(ctx, refreshInterval)
+       }
 
        sys.printIAMRoles()
 
diff --git a/cmd/server-main.go b/cmd/server-main.go
index de0ff1898..7be6eac87 100644
--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -972,7 +972,7 @@ func serverMain(ctx *cli.Context) {
        }
 
        // Initialize users credentials and policies in background right after config has initialized.
-       globalIAMFullyInitialized.Store(true)
+       //globalIAMFullyInitialized.Store(true)
        go func() {
                bootstrapTrace("globalIAMSys.Init", func() {
                        globalIAMSys.Init(GlobalContext, newObject, globalEtcdClient, globalRefreshIAMInterval)
```

Reply:
ok, it looked like the `// Initialize Console UI` was outside this goroutine.

Having the duplicated Console UI init seems weird.

A) It would already have been init'ed outside this goroutine? Why init it twice? Is that safe?
B) If we can initConsoleServer even if IAM hasn't loaded, why wait for it to try once?